src/boot/efi/boot.c

```diff
@@ -741,23 +741,21 @@ static bool menu_run(
                         lines = xnew(char16_t *, config->n_entries + 1);
 
                         for (size_t i = 0; i < config->n_entries; i++) {
-                                size_t j, padding;
-
-                                lines[i] = xnew(char16_t, line_width + 1);
-                                padding = (line_width - MIN(strlen16(config->entries[i]->title_show), line_width)) / 2;
+                                int width = line_width - MIN(strlen16(config->entries[i]->title_show), line_width);
```

Comment:
please be careful with types. i.e. the types involved here are all size_t (i.e. unsigned 64bit), but you assign this to an int (i.e. signed 32bit). please don't will-nilly overflow this by accident, i.e. stick to size_t

Reply:
The issue is that xasprintf expects int and only int (but that's based on stackoverflow answers), so I have to cast them to int at some point.
That's why I converted them to int, especially since it's imo unrealistic that any of these values get bigger than INT_MAX
If wanted, I can add checks to prevent overflows or do you know a better solution?

Reply:
yeah printf() is a bit weak on that, they should have used size_t too. But i guess in the olden times they weren't really so strong in getting the types right.

but please, let's stay in the right types as long as we can, and only cast this the moment we pass this over to xasprintf(), not any earlier.

And consider adding "assert(… <= INT_MAX)" first, so that the assumption about the ranges is made explicit rather than implicit.

Reply:
done :)